Code/Tools/ProjectManager/Source/GemCatalog/GemModel.h

```diff
@@ -53,20 +51,32 @@ namespace O3DE::ProjectManager
         static QString GetRequirement(const QModelIndex& modelIndex);
 
         static bool IsAdded(const QModelIndex& modelIndex);
+        static bool IsAddedDependency(const QModelIndex& modelIndex);
         static void SetIsAdded(QAbstractItemModel& model, const QModelIndex& modelIndex, bool isAdded);
+        static void SetIsAddedDependency(QAbstractItemModel& model, const QModelIndex& modelIndex, bool isAdded);
         static void SetWasPreviouslyAdded(QAbstractItemModel& model, const QModelIndex& modelIndex, bool wasAdded);
-        static bool NeedsToBeAdded(const QModelIndex& modelIndex);
-        static bool NeedsToBeRemoved(const QModelIndex& modelIndex);
+        static bool WasPreviouslyAdded(const QModelIndex& modelIndex);
+        static void SetWasPreviouslyAddedDependency(QAbstractItemModel& model, const QModelIndex& modelIndex, bool wasAdded);
+        static bool WasPreviouslyAddedDependency(const QModelIndex& modelIndex);
+        static bool NeedsToBeAdded(const QModelIndex& modelIndex, bool includeDependencies = false);
+        static bool NeedsToBeRemoved(const QModelIndex& modelIndex, bool includeDependencies = false);
         static bool HasRequirement(const QModelIndex& modelIndex);
+        static void UpdateDependencies(QAbstractItemModel& model, const QModelIndex& modelIndex);
 
         bool DoGemsToBeAddedHaveRequirements() const;
 
-        QVector<QModelIndex> GatherGemsToBeAdded() const;
-        QVector<QModelIndex> GatherGemsToBeRemoved() const;
+        QVector<QModelIndex> GatherGemDependencies(const QModelIndex& modelIndex);
+        QVector<QModelIndex> GatherDependentGems(const QModelIndex& modelIndex, bool addedOnly = false) const;
+        QVector<QModelIndex> GatherGemsToBeAdded(bool includeDependencies = false) const;
+        QVector<QModelIndex> GatherGemsToBeRemoved(bool includeDependencies = false) const;
 
-        int TotalAddedGems() const;
+        int TotalAddedGems(bool includeDependencies = false) const;
 
     private:
+        void FindGemDisplayNamesByNameStrings(QStringList& inOutGemNames);
+        void GetAllDependingGems(const QModelIndex& modelIndex, QSet<QModelIndex>& inOutGems);
```

Comment:
Name didn't make me think this would get the whole gem dependency chain of gems and not just the first order dependencies. Can't think of a better name for the function that specifies that though so I don't blame you if you leave it as is. Maybe add a comment to the declaration?

Reply:
I'm looking into moving this function to python so it may not exist here soon.

Reply:
OK, looking at moving this work to another PR - tracking with issue https://github.com/o3de/o3de/issues/4301